Terminal.Gui/Views/GraphView.cs

```diff
@@ -0,0 +1,1913 @@
+			for (int i = 0; i < Bars.Count; i++) {
+
+				decimal barPosition = (i + 1) * BarEvery;
+				barPosition += Offset;
+
+				// the x/y position that the cell would have to be between for the bar to be rendered
+				var low = Orientation == Orientation.Vertical ? graphSpace.X : graphSpace.Y;
+				var high = Orientation == Orientation.Vertical ? graphSpace.Right : graphSpace.Bottom;
+
+				// if a bar contained in this cell's X/Y axis of data space
+				if (barPosition >= low && barPosition < high) {
+					return Bars [i];
+				}
+			}
+
+			return null;
+		}
+
+		/// <summary>
+		/// Returns the name of the bar (if any) that is rendered at this
+		/// point in the x axis
+		/// </summary>
+		/// <param name="axisPoint"></param>
+		/// <returns></returns>
+		public string GetLabelText (AxisIncrementToRender axisPoint)
+		{
+			return LocationToBar (axisPoint.GraphSpace)?.Text;
+		}
+
+		/// <summary>
+		/// A single bar in a <see cref="BarSeries"/>
+		/// </summary>
+		public class Bar {
+
+			/// <summary>
+			/// Optional text that describes the bar.  This can be added as a label on the axis by setting
+			/// <see cref="Axis.LabelGetter"/> = <see cref="BarSeries.GetLabelText(AxisIncrementToRender)"/>
+			/// </summary>
+			public string Text { get; set; }
+
+			/// <summary>
+			/// The color and character that will be rendered in the console
+			/// when the bar extends over it
+			/// </summary>
+			public GraphCellToRender Fill { get; set; }
+
+			/// <summary>
+			/// Allows you to control the color of the bar at a given graph location.  This
+			/// overrides the <see cref="Fill"/>
+			/// </summary>
+			public GraphAttributeGetterDelegate ColorGetter { get; set; }
+
+			/// <summary>
+			/// The value in graph space X/Y (depending on <see cref="Orientation"/>) to which the bar extends.
+			/// </summary>
+			public decimal Value { get; }
+
+			/// <summary>
+			/// Creates a new instance of a single bar rendered in the given <paramref name="fill"/> that extends
+			/// out <paramref name="value"/> graph space units in the default <see cref="Orientation"/>
+			/// </summary>
+			/// <param name="text"></param>
+			/// <param name="fill"></param>
+			/// <param name="value"></param>
+			public Bar (string text, GraphCellToRender fill, decimal value)
+			{
+				Text = text;
+				Fill = fill;
+				Value = value;
+			}
+
+			/// <summary>
+			/// Returns <see cref="Fill"/> with optional overriding color (see <see cref="ColorGetter"/>)
+			/// </summary>
+			/// <returns></returns>
+			internal GraphCellToRender GetFinalFill (RectangleD graphSpace)
+			{
+				var customColor = ColorGetter?.Invoke (graphSpace);
+
+				if (customColor == null) {
+					return Fill;
+				}
+
+				return new GraphCellToRender (Fill.Rune, customColor);
+			}
+		}
+	}
+
+	/// <summary>
+	/// Sequence of lines to connect points e.g. of a <see cref="ScatterSeries"/>
+	/// </summary>
+	public class PathAnnotation : IAnnotation {
```

Comment:
I'm being repetitive here (and going from bottom up, so sorry about that).

If some `Terminal.Gui` user encounters this class in the object browser or documentation they will have no clue what it is for. It may distract them from their primary goal. Simillarly, someone LOOKING for this functionality in relation to `GraphView` will not know it is part of `GraphView`.

There are many ways of addressing this. 

- Rename the class to include `GraphView` (ugh)
- Create a namespace as a container
- Refactor some  other way

What other solutions are there? (I'm sincerely asking).

Reply:
It could just get its own namespace.

That means that only those using graphics would bring the additional types.

Reply:
I think changing the namespace to `namespace Terminal.Gui.GraphView` and put inside a folder named `GraphView`  may be a solution.

Reply:
Thanks, I will move them to a sub namespace and folder and look to see if any could be made private.

Reply:
Since it is to stay on a different namespace/folder and if these classes needs to be public for derived classes or whatever, I think they can stay as they are.

Reply:
I now saw the structure and I think the `Terminal.Gui.Graphs` should be inside the `Core` folder with a `Graph .cs` file with all the APIs that belong to him. The `GraphView.cs` file could be stay on the root of the `Views` folder with only the APIs of the view. What do you think?